core/src/test/java/org/mobilitydata/gtfsvalidator/table/AnyTableLoaderTest.java

```diff
@@ -0,0 +1,239 @@
+package org.mobilitydata.gtfsvalidator.table;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mobilitydata.gtfsvalidator.TestUtils.toInputStream;
+import static org.mobilitydata.gtfsvalidator.TestUtils.validationNoticeTypes;
+import static org.mockito.Mockito.*;
+
+import com.google.common.collect.ImmutableList;
+import java.io.InputStream;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.mobilitydata.gtfsvalidator.annotation.FieldLevelEnum;
+import org.mobilitydata.gtfsvalidator.input.CountryCode;
+import org.mobilitydata.gtfsvalidator.input.CurrentDateTime;
+import org.mobilitydata.gtfsvalidator.notice.*;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsTestEntity;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsTestFileValidator;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsTestTableContainer;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsTestTableDescriptor;
+import org.mobilitydata.gtfsvalidator.validator.DefaultValidatorProvider;
+import org.mobilitydata.gtfsvalidator.validator.ValidationContext;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorLoader;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorProvider;
+
+public class AnyTableLoaderTest {
+
+  ValidationContext validationContext;
+
+  @Before
+  public void setup() {
+    validationContext =
+        ValidationContext.builder()
+            .setCountryCode(CountryCode.forStringOrUnknown("CA"))
+            .setCurrentDateTime(new CurrentDateTime(ZonedDateTime.now(ZoneId.systemDefault())))
+            .build();
+  }
+
+  @Test
+  public void invalidInputStream() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("_not_a_valid_file_");
+    GtfsTableContainer mockContainer = mock(GtfsTableContainer.class);
+    when(testTableDescriptor.createContainerForInvalidStatus(
+            GtfsTableContainer.TableStatus.INVALID_HEADERS))
+        .thenReturn(mockContainer);
+    NoticeContainer loaderNotices = new NoticeContainer();
+
+    var loadedContainer =
+        AnyTableLoader.load(
+            testTableDescriptor, mock(ValidatorProvider.class), null, loaderNotices);
+
+    assertThat(loaderNotices.hasValidationErrors()).isTrue();
```

Comment:
Echoing my previous comment, I'd still argue for dropping this `hasValidationErrors` check, both here and below.  It's duplicative of the next check, but if it fails, it doesn't give you as much information as the `validationNoticeTypes` check on the next line.  Thus, favor the next check.